src/js/pm-image-editor.js

```diff
@@ -444,6 +448,11 @@
                     $scope.$broadcast('updateSelection', $scope.editorId, $scope.editor.selectionCss());
 
                     $scope.updateHistoryButtons();
+
+                    $scope.state = angular.copy($scope.editor.history.current());
+                    setTimeout(function() {
+                        $scope.$apply();
+                    }, 0);
```

Comment:
Why do you keep adding random timeouts? (I know that `setTimeout 0` is a legitimate JS pattern, but it's not needed here.) This is 3rd time recently...

Other places:
https://github.com/partnermarketing/pm2-frontend/pull/1018#issuecomment-182793511
https://github.com/partnermarketing/pm2-frontend/pull/1009

Also, this doesn't catch errors if there is a $digest or $apply already is progress.

Why call $apply manually anyway?